sdk/core/azure-mgmt-core/azure/mgmt/core/policies/__init__.py

```diff
@@ -24,9 +24,28 @@
 #
 # --------------------------------------------------------------------------
 
+from azure.core.pipeline.policies import HttpLoggingPolicy
 from ._base import ARMAutoResourceProviderRegistrationPolicy
 
-__all__ = ["ARMAutoResourceProviderRegistrationPolicy"]
+
+class ARMHttpLoggingPolicy(HttpLoggingPolicy):
+    """HttpLoggingPolicy with ARM specific safe headers fopr loggers.
+    """
+
+    DEFAULT_HEADERS_WHITELIST = HttpLoggingPolicy.DEFAULT_HEADERS_WHITELIST | set([
+        # https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling#remaining-requests
+        "x-ms-ratelimit-remaining-subscription-reads",
+        "x-ms-ratelimit-remaining-subscription-writes",
+        "x-ms-ratelimit-remaining-tenant-reads",
+        "x-ms-ratelimit-remaining-tenant-writes",
+        "x-ms-ratelimit-remaining-subscription-resource-requests",
+        "x-ms-ratelimit-remaining-subscription-resource-entities-read",
+        "x-ms-ratelimit-remaining-tenant-resource-requests",
+        "x-ms-ratelimit-remaining-tenant-resource-entities-read",
```

Comment:
Compute has a bunch of other headers as well. 

For example, I don't see `x-ms-ratelimit-remaining-resource`

Reply:
I used https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/request-limits-and-throttling#remaining-requests is there anything else I should follow?

Reply:
There is also https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/troubleshooting-throttling-errors#call-rate-informational-response-headers

Reply:
done, anything else?

Reply:
Why do we add headers for compute here, but not in compute SDK? @johan.stenberg@microsoft.com 

---
In reply to: [412493196](https://github.com/Azure/azure-sdk-for-python/pull/10940#discussion_r412493196) [](ancestors = 412493196)

Reply:
I'd be fine with adding the compute specific header in the compute library. Adding it to the azure-mgmt-core is significantly less broad than adding it to azure-core, however - the probability that another resource provide would add the header *and* that it would be sensitive is fast approaching nil. Add to this the fact that the compute library is autogenerated, and thus would need a way to inject the additional policies, it may cost more than what value we get out of moving the header into the compute specific package.

Reply:
Your points make sense to me.

---
In reply to: [413154247](https://github.com/Azure/azure-sdk-for-python/pull/10940#discussion_r413154247) [](ancestors = 413154247)